internal/component/prometheus/operator/types.go

```diff
@@ -24,6 +24,8 @@ type Arguments struct {
 	// Namespaces to search for monitor resources. Empty implies All namespaces
 	Namespaces []string `alloy:"namespaces,attr,optional"`
 
+	KubernetesRole string `alloy:"kubernetes_role,attr,optional"`
```

Comment:
We will need to document this in user-facing docs:

- https://github.com/grafana/alloy/blob/main/docs/sources/reference/components/prometheus/prometheus.operator.podmonitors.md
- https://github.com/grafana/alloy/blob/main/docs/sources/reference/components/prometheus/prometheus.operator.servicemonitors.md
- https://github.com/grafana/alloy/blob/main/docs/sources/reference/components/prometheus/prometheus.operator.probes.md